ea-2021-04-ndvi-automation_haselwimmer.py

```diff
@@ -0,0 +1,663 @@
+# 
+
+# ### Add Your Name Below 
+# **Your Name: Christian Haselwimmer**
+
+# <img style="float: left;" src="colored-bar.png"/>
+
+# ---
+
+# # Week 04 and 05 Homework - Automate NDVI Workflow
+# 
+# For this assignment, you will write code to generate a plot of the mean normalized difference vegetation index (NDVI) for two different sites in the United States across one year of data:
+# 
+# * San Joaquin Experimental Range (SJER) in Southern California, United States
+# * Harvard Forest (HARV) in the Northeastern United States
+# 
+# The data that you will use for this week is available from **earthpy** using the following download: 
+# 
+# `et.data.get_data('ndvi-automation')`
+# 
+# ## Assignment Goals
+# 
+# Your goal in this assignment is to create the most efficient and concise workflow that you can that allows for:
+# 
+# 1. The code to scale if you added new sites or more time periods to the analysis.
+# 2. Someone else to understand your workflow.
+# 3. The LEAST and most efficient (i.e. runs fast, minimize repetition) amount of code that completes the task.
+# 
+# ### HINTS
+# 
+# * Remove values outside of the landsat valid range of values as specified in the metadata, as needed.
+# * Keep any output files SEPARATE FROM input files. Outputs should be created in an outputs directory that is created in the code (if needed) and/or tested for.
+# * Use the functions that we demonstrated during class to make your workflow more efficient.
+# * BONUS - if you  chose - you can export your data as a csv file. You will get bonus points for doing this.
+# 
+# 
+# ## Assignment Requirements
+# 
+# Your submission to the GitHub repository should include:
+# * This Jupyter Notebook file (.ipynb) with:
+#     * The code to create a plot of mean NDVI across a year for  2 NEON Field Sites:
+#         * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object
+#     * The **data should be cleaned to remove the influence of clouds**. See the [earthdatascience website for an example of what your plot might look like with and without removal of clouds](https://www.earthdatascience.org/courses/earth-analytics-python/create-efficient-data-workflows/).
+# * BONUS: Create one output `.csv` file that has 3 columns - NDVI, Date and Site Name - with values for SJER and HARV.
+# 
+# Your notebook should:
+# * Have *at least* 2 well documented and well named functions with docstrings.
+# * Include a Markdown cell at the top of the notebook that outlines the overall workflow using pseudocode (i.e. plain language, not code)
+# * Include additional Markdown cells throughout the notebook to describe: 
+#     * the data that you used - and where it is from
+#     * how data are being processing
+#     * how the code is optimized to run fast and be more concise
+
+# # Replace this cell with your pseudocode  for this workflow
+# 
+# If you happen to be a diagram person a diagram is ok too
+# 
+# 
+
```

Comment:
@haselwimmer Looks great Christian! Really appreciated seeing some of your creative solutions. Code looks good. Comments throughout were clear and helpful. Nice job!